src/Solarium/EventDispatcher/Psr14Bridge.php

```diff
@@ -20,7 +19,18 @@ public function __construct(ContainerAwareEventDispatcher $eventDispatcher) {
     $this->dispatcher = $eventDispatcher;
   }
 
-  public function dispatch($event, Event $null = NULL) {
+  /**
+   * Call magic method to account for Symfony\Contracts vs Symfony\Component method declarations.
+   */
+  public function __call($name, $args) {
```

Comment:
I don't super love this but it works. I'm open to suggestions :)

Reply:
We could just drop compatibility with the older version and bump the core requirement.

Reply:
Per https://www.drupal.org/node/3159012 this change was introduced in 9.1.0.